Profiles - ZIB 2017/zib-SkinDisorder.xml

```diff
@@ -0,0 +1,167 @@
+  <copyright value="CC0" />
+  <fhirVersion value="3.0.1" />
+  <mapping>
+    <identity value="SkinDisorder-v3.2-2017EN" />
+    <uri value="https://zibs.nl/wiki/SkinDisorder-v3.2(2017EN)" />
+    <name value="SkinDisorder-v3.2(2017EN)" />
+  </mapping>
+  <kind value="resource" />
+  <abstract value="false" />
+  <type value="Condition" />
+  <baseDefinition value="http://nictiz.nl/fhir/StructureDefinition/zib-Problem" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="Condition.extension">
+      <path value="Condition.extension" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="url" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+    </element>
+    <element id="Condition.extension:dueTo">
+      <path value="Condition.extension" />
+      <sliceName value="dueTo" />
+      <short value="Cause" />
+      <definition value="The cause of the problem underlying the skin disorder." />
+      <alias value="Oorzaak" />
+      <min value="0" />
+      <max value="1" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.org/fhir/StructureDefinition/zib-SkinDisorder-dueTo" />
+      </type>
+    </element>
+    <element id="Condition.extension:dueTo.valueReference:valueReference">
+      <path value="Condition.extension.valueReference" />
+      <sliceName value="valueReference" />
+      <mapping>
+        <identity value="SkinDisorder-v3.2-2017EN" />
+        <map value="NL-CM:19.3.7" />
+      </mapping>
+    </element>
+    <element id="Condition.category.coding">
+      <path value="Condition.category.coding" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="code" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+    </element>
+    <element id="Condition.category.coding:SkinDisorderCoding">
+      <path value="Condition.category.coding" />
+      <sliceName value="SkinDisorderCoding" />
+    </element>
+    <element id="Condition.category.coding:SkinDisorderCoding.system">
+      <path value="Condition.category.coding.system" />
+      <fixedUri value="http://snomed.info/sct" />
+    </element>
+    <element id="Condition.category.coding:SkinDisorderCoding.code">
+      <path value="Condition.category.coding.code" />
+      <fixedCode value="95320005" />
+    </element>
+    <element id="Condition.code">
+      <path value="Condition.code" />
+      <short value="TypeOfDisorder" />
+    </element>
+    <element id="Condition.code.coding">
+      <path value="Condition.code.coding" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="code" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+      <min value="1" />

```

Comment:
Yes I agree, but since .code is required per spec, I assume if a code is required, then please use coding of TypeOfDisorder. This is not as the zib dictates, but this is more implementable, since a code is required.

Do you agree?

Reply:
Well ... this is actually a pretty tough question. It's ultimately `.category` in this case that identifies the kind of observation, while `.code` only further specifies it. The zib mentions explicitly: 'we purposefully opted for a limited list. The reason for this is that there are many types of skin conditions that are almost always recorded as free text'. So the intention of the list predominantly seems to be to record free text(!).
Normally the text field of the CodeableConcept type seems to be best suited for this purpose. However, the zib mentions that this should be done through the use of the OTH NullFlavor code. So what's best? FHIR specifies that at least a .code is needed, the zib makes it in effect completely optional how this should be filled in (you MAY use this list, and if you use it, you MAY use an ordinary text string). In the ideal situation, staying true to the zib *and* to FHIR. the OTH would actually be replaced with the text field of the CodeableConcept type of `.code` and make a constraint that (at least) either a coding or the text is present.
But since we have to deal with the given valueset, maybe it's indeed best to bind it 1..1? Maybe Ardon can chime in as well?

Reply:
@pieter-edelman-nictiz , @furoreSebastiaan ,

Not the FHIR spec requires the code 1..1 but our HCIM Problem!

I am in favour of making the code.coding 0..*  so the code.text could be used as well and we fall a little bit more in line with the HCIM.

And perhaps we should discuss if we should use HCIM problem as a base.

Reply:
I think a code is pretty essential and there is an escape possibility with the OTH or using the code..text. Therefore I think we can use the HCM Problem as a base.  

Updated the profile and will merge to develop. Lets discuss in real life.